app/controllers/application_controller.rb

```diff
@@ -178,11 +178,13 @@ def prioritize_json_format
   # rubocop:enable Metrics/PerceivedComplexity
 
   def redirect_location
-    next_redirect_chain_value || session.delete('user_return_to') || root_path
+    next_redirect_chain_value ||
+      session.delete('user_return_to') ||
+      root_path
```

Comment:
(I was going to add a fourth option, but didn't end up doing so, but thought it's easier to read split out, anyway, so I'm leaving this change.)